src/pytest_bdd/parser.py

```diff
@@ -521,18 +522,18 @@ def _parse_and_add_rule(self, rule_data: GherkinRule, feature: Feature) -> None:
 
     def _extract_rule_background(self, rule_data: GherkinRule) -> Background | None:
         """Extract the first background from rule children if it exists."""
-        for child in rule_data.children:
-            if child.background:
-                return self.parse_background(child.background)
+        for child in rule_data["children"]:
+            if "background" in child:
+                return self.parse_background(child["background"])  # type: ignore[typeddict-item]
         return None
 
     def _extract_rule_scenarios(
         self, rule_data: GherkinRule, feature: Feature, rule: Rule
     ) -> Generator[ScenarioTemplate]:
         """Yield each parsed scenario under a rule."""
-        for child in rule_data.children:
-            if child.scenario:
-                yield self.parse_scenario(child.scenario, feature, rule)
+        for child in rule_data["children"]:
+            if "scenario" in child:
+                yield self.parse_scenario(child["scenario"], feature, rule)  # type: ignore[typeddict-item]
```

Comment:
we should try avoiding `# type: ignore`. What's the issue here? Maybe we need to cast `child` to a specific type?